components/server/src/workspace/workspace-starter.ts

```diff
@@ -244,6 +248,12 @@ export async function getWorkspaceClassForInstance(
     }
 }
 
+class StartInstanceError extends Error {
+    constructor(public readonly reason: FailedInstanceStartReason, public readonly cause: Error) {
+        super("Starting workspace instance failed: " + cause.message);
```

Comment:
How is the reason value assigned to this inherited type before it is used below?
 
```
if (err instanceof StartInstanceError) {
    failedReason = err.reason;
}
```
I don't have that much experience with Errors in JS/TS but here it looks like we're just assigning it to a possibly variable string. My worry with this is that for the metric in prometheus, we should ideally have a fixed cardinality of the label which means we wouldn't expect the variance that could come from this constructor taking on arbitrary `message`.

Reply:
The [type](https://github.com/gitpod-io/gitpod/pull/12378/files#diff-fc6040e0ca736fbc219a7234a6c4e6c1c9dfbf711210358df76dd57f45530532R151) is restricted to only three values.